Cherry-pick #29877 : Use GetAllByCap to get list of network plugins in docker info

mavenugo opened the pull request:
Cherry picking #29877 to 1.13 branch as well.

returned in "docker info". Currently info endpoint isnt using the
GetAllByCap, but relies on existing networks to get the plugin names.
This causes a basic issue when it comes to global network plugins which
swarm-mode relies on, wherein swarmkit will not be able to schedule the
network on the worker nodes due to the filtering logic.

In order to break this chicken & egg issue, we must start to use the
GetAllManagedPluginsByCap. We are unable to use GetAllByCap due to
various issues with Plugin-V1's lazy loading approach causing issues
especially during daemon restarts (which uses SystemInfo)

Signed-off-by: Madhu Venugopal <madhu@docker.com>

thaJeztah commented:
CI looks to be failing;

```
13:16:18 --- FAIL: Test (3265.54s)
13:16:18 panic: DockerSwarmSuite.TestSwarmNetworkPlugin test timed out after 5m0s [recovered]
13:16:18 	panic: DockerSwarmSuite.TestSwarmNetworkPlugin test timed out after 5m0s
13:16:18 
13:16:18 goroutine 8 [running]:
13:16:18 panic(0x12257a0, 0xc42063e010)
13:16:18 	/usr/local/go/src/runtime/panic.go:500 +0x1a1
13:16:18 testing.tRunner.func1(0xc42007a840)
13:16:18 	/usr/local/go/src/testing/testing.go:579 +0x25d
13:16:18 panic(0x12257a0, 0xc42063e010)
...
```

mavenugo commented:
@thaJeztah this will be solved by https://github.com/docker/docker/pull/29733 (once it is cherry-picked into 1.13.x branch). cc @cpuguy83 @vieux

mavenugo committed:
Swarmkit has enabled the plugin filtering logic based on the drivers
returned in "docker info". Currently info endpoint isnt using the
GetAllByCap, but relies on existing networks to get the plugin names.
This causes a basic issue when it comes to global network plugins which
swarm-mode relies on, wherein swarmkit will not be able to schedule the
network on the worker nodes due to the filtering logic.

In order to break this chicken & egg issue, we must start to use the
GetAllManagedPluginsByCap. We are unable to use GetAllByCap due to
various issues with Plugin-V1's lazy loading approach causing issues
especially during daemon restarts (which uses SystemInfo)

Signed-off-by: Madhu Venugopal <madhu@docker.com>

vieux commented:
LGTM

tiborvass commented:
LGTM
